src/middlewared/middlewared/plugins/mail.py

```diff
@@ -441,7 +440,7 @@ def send_mail_queue(self):
             for queue in list(mq.queue):
```

Comment:
I'm just making a comment here and not asking it to be changed in this PR, but we should consider changing `_get_smtp_server` to a context manager and use the `with SMTP**` context manager so we actually release resources properly in event of tracebacks.

Reply:
I'll do it in 25.04 after we merge and backport this